spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/channel/PostgresSubscribableChannel.java

```diff
@@ -71,6 +73,8 @@ public class PostgresSubscribableChannel extends AbstractSubscribableChannel
 
 	private final UnicastingDispatcher dispatcher = new UnicastingDispatcher();
 
+	private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
```

Comment:
Let's make its name more specific to its usage!
How about `hasHandlersLock`?

Reply:
Renamed.